core/auth/fake/identifier.go

```diff
@@ -0,0 +1,236 @@
+}
+
+// Inject injects module dependencies
+func (i *Identifier) Inject(
+	reverseRouter web.ReverseRouter,
+	responder *web.Responder,
+	eventRouter flamingo.EventRouter,
+) *Identifier {
+	i.reverseRouter = reverseRouter
+	i.responder = responder
+	i.eventRouter = eventRouter
+
+	return i
+}
+
+// Broker returns the broker id from the config
+func (i *Identifier) Broker() string {
+	return i.broker
+}
+
+// Authenticate action, fake
+func (i *Identifier) Authenticate(_ context.Context, r *web.Request) web.Result {
+	var formError error
+
+	postValues, err := r.FormAll()
+	if err == nil {
+		delete(postValues, "broker")
+		if len(postValues) > 0 {
+			formError = i.handlePostValues(r, postValues, i.broker)
+
+			if formError == nil {
+				return i.responder.RouteRedirect("core.auth.callback", map[string]string{"broker": i.broker})
+			}
+		}
+	}
+
+	// get formURL to callback with broker filled in
+	formURL, err := i.reverseRouter.Absolute(r, "core.auth.login", map[string]string{"broker": i.broker})
+	if err != nil {
+		return i.responder.ServerError(err)
+	}
+
+	var loginTemplate string
+	if i.config.LoginTemplate != "" {
+		loginTemplate = i.config.LoginTemplate
+	} else {
+		loginTemplate = defaultLoginTemplate
+	}
+
+	t := template.New("fake")
+	t, err = t.Parse(loginTemplate)
+	if err != nil {
+		return i.responder.ServerError(err)
+	}
+
+	var body = new(bytes.Buffer)
+	var errMsg string
+
+	if formError != nil {
+		errMsg = formError.Error()
+	}
+
+	err = t.Execute(
+		body,
+		viewData{
+			FormURL:    formURL.String(),
+			Message:    errMsg,
+			UsernameID: i.config.UsernameFieldID,
+			PasswordID: i.config.PasswordFieldID,
+		})
+	if err != nil {
+		return i.responder.ServerError(err)
+	}
+
+	return &web.Response{
+		Header: http.Header{"ContentType": []string{"text/html; charset=utf-8"}},
+		Status: http.StatusOK,
+		Body:   body,
+	}
+}
+
+func (i *Identifier) handlePostValues(r *web.Request, values map[string][]string, broker string) error {
+	usernameVal, ok := values[i.config.UsernameFieldID]
+	if !ok {
+		return errMissingUsername
+	}
+
+	user := usernameVal[0]
+
+	userCfg, found := i.config.UserConfig[user]
+	if !found {
+		return errInvalidUser
+	}
+
+	if i.config.ValidatePassword {
+		passwordVal, ok := values[i.config.PasswordFieldID]
+		if !ok {
+			return errMissingPassword
+		}
+
+		expectedPassword := passwordVal[0]
+		userPassword := userCfg.Password
+		if expectedPassword != userPassword {
+			return errPasswordMismatch
+		}
+	}
+
+	sessionData := UserSessionData{Subject: user}
+	r.Session().Store(fmt.Sprintf(userDataSessionKey, broker), sessionData)
+
+	return nil
+}
+
+// Identify action, fake
+func (i *Identifier) Identify(ctx context.Context, request *web.Request) (auth.Identity, error) {
+	userSessionData, ok := request.Session().Load(fmt.Sprintf(userDataSessionKey, i.broker))
+	if !ok {
+		return nil, errors.New("identity not saved in session")
```

Comment:
Make this error a `errNoIdentity` variable to be consistent with the other errors?